themes/bootstrap/static/js/filter.js

```diff
@@ -0,0 +1,228 @@
+"use strict";
+
+let excludeDormant = false;
+let excludeArchived = true;
+let searchTerm = '';
+const excludeDormantCheckbox = document.getElementById('exclude-dormant');
+const includeArchivedCheckbox = document.getElementById('include-archived');
+const searchResultMessage = document.getElementById('search-result-message');
+let sortedTools = [];
+let owners = new Set();  // owners and languages become arrays after filling.
+let languages = new Set();
+const selectedLanguages = new Set();
+const selectedOwners = new Set();
+
+
+/** Load the tools that are not archived. */
+$.getJSON('../tools-data.json', function(toolsData) {
+  sortedTools = sortTools(toolsData);
+
+  languages = populateLanguages(sortedTools);
+  owners = populateOwners(sortedTools);
+
+  populateToolsKeywords(sortedTools);
+  fillToolsTable(sortedTools, selectedLanguages, selectedOwners);
+  initFormProcessors();
+});
+
+
+/** Given the tools data, return it sorted by rating and name. */
+function sortTools(toolsData) {
+  const sortedTools = toolsData.sort((a, b) => {
+      if (b.rating === a.rating) {
+          return a.name.localeCompare(b.name);
+      }
+      return b.rating - a.rating;
+  });
+  return sortedTools;
+}
+
+/** Given the tools data, return a sorted array of the languages
+ *  to be used for filtering.
+ */
```

Comment:
Use [Google JavaScript code style](https://google.github.io/styleguide/jsguide.html#jsdoc-general-form) 
```suggestion
/** 
 * Given the tools data, return a sorted array of the languages to be used for filtering.
 */
```
In general, I would format all JSDoc like this (i.e. also those in line 16, 29, ...) and rather use _one sentence, one line_ than _hard wrap at X chars_...